tests/main/snap-set-core-config/task.yaml

```
@@ -4,19 +4,16 @@ summary: Ensure "snap set core" works
 systems: [ubuntu-core-*]
 
 prepare: |
-    if systemctl -l | MATCH rsyslog.service; then
-        echo "rsyslog service exists on the system"
-        exit 0
-    fi
-
-    #shellcheck source=tests/lib/systemd.sh
-    . "$TESTSLIB"/systemd.sh
+    if ! systemctl -l | MATCH rsyslog.service; then
```

Should this be spelled out like this instead?

```suggestion
    If not systemctl -l | MATCH rsyslog.service; then
```

---

@zyga thanks for the review. I already tried the not command but it doesn't work like using !
I didn't go deep on why it does not work yet.

---

I think `not` will not work here because it negates the exit status of it's direct subcommand, in this case it's MATCH that fails. I think you could use
```
if systemctl -l | MATCH -v rsyslog.service ; then
    # no rsyslog service, prepare a fake one
    ....
fi
```

---

@bboozzoo thanks for the proposal. It is already pushed